### DISCUSSION TITLE
Rust Documentation

### DISCUSSION THREAD

**Discussion by Martyx00** [Vector35/binaryninja-api#3181] [API] [answered]
Hello,
I would like to ask if there is any existing draft of Rust API documentation. I can live with the fact that it is not complete and that it may change, however, I have failed to find any.

Thanks :)

**Comment by fuzyll** [comment, marked as answer]
Hey, sorry about that! The documentation is meant to be used like a standard Rust crate. If you've got this repository checked out locally, you'll want to do:

```
cd rust
cargo doc --open
```

That should build the documentation for the Rust bindings and then open it in your browser.

If you have any issues with that, or have any problems with the bindings themselves, please let us know!

**Reply by Martyx00** [reply]
It is likely just me being retarded (I know just basics of Rust), but would you have some example of registering own custom architecture? I just cant make it work. Thanks :)
